### PR TITLE
Remove assert_raises_regex (nose dependent)

### PR DESCRIPTION
**Pull request by kchawla-pi**
`assert_raises_regex` is a wrapper for a nose.tools function. WIthout replacing it with pytest.raises, we cannot remove `nose` from the CT.

### CODE REVIEW

**Comment by jeromedockes** [comment]
why is `assert_raises_regex` a problem?

**Comment by kchawla-pi** [comment]
Its a wrapper for a `nose.tools` function. I tried removing nose from the CI and it started failing.

**Comment by kchawla-pi** [comment]
Corrected the PR title.

**Comment by kchawla-pi** [comment]
This one won't need a review I think.

**Comment by jeromedockes** [comment]
> Its a wrapper for a `nose.tools` function. I tried removing nose from the CI and it started failing.

thanks, good to know!

**Comment by jeromedockes** [comment]
> This one won't need a review I think.

I looked at it earlier, LGTM!

File: nilearn/surface/tests/test_surface.py [hunk 1]
```diff
@@ -7,17 +7,13 @@
...
 
+import nibabel as nb
+import numpy as np
 import pytest
 
 from numpy.testing import assert_array_equal, assert_array_almost_equal
-
-from nilearn._utils.testing import assert_raises_regex, assert_warns
-
-import numpy as np
```
Comment (bthirion) [inline nilearn/surface/tests/test_surface.py]: ts ts ts this should be in another PR ;-)

Comment (kchawla-pi) [reply nilearn/surface/tests/test_surface.py]: I know!!! And yet...
